### DISCUSSION TITLE
Deprecations for v10

### DISCUSSION THREAD

**Discussion by mfazekas** [rnmapbox/maps#2627] [Ideas]
I'm planning the following deprecations in v10. So we can remove them in after we've released v10 final:

## Deprecated implementations MapLibre and MapboxGL

MapLibre and classic MapoxGL implementations will be deprecated. Still usable to aid people in conversions, but will be removed after 10.0 is released.

## Images 

<h3 id="image-in-style-deprecated" style="{marginTop: 60px}"> using images/urls directly in styles</h3>

#### Deprecated:

   ```jsx
   <SymbolLayer style={{iconImage: require('foo.png')}} />
   ```

   ```jsx
   <SymbolLayer style={{iconImage: 'https://download.samplelib.com/png/sample-red-100x75.png'}} />
   ```

#### Replacement:

   ```jsx
   <Images images={{foo: require('foo.png')}} />
   <SymbolLayer style={{iconImage: 'foo'}}/>
   ```

   ```jsx
   <Images images={{foo: 'sample-red-100x75.png'}} />
   <SymbolLayer style={{'sample-red-100x75':   'https://download.samplelib.com/png/sample-red-100x75.png'}}/>
   ```

#### Why?

Required a lot of extra code, and it's esp. an issue since image loading is async. The alternative is more flexible.

### ~`onRegionDidChange`~ => `onMapIdle` , ~`onRegionIsChanging`~ =>  `onCameraChanged

will be deprecated, in v10 you can use `onMapIdle` and `onCameraChanged` names are much better.

#### Deprecated:

   ```jsx
   <SymbolLayer><View style={{backgroundColor:'red', width: 20, height: 20, borderRadius: 20}} /></SymbolLayer>
   ``` 
   
#### Replacement:

   ```jsx
   <SymbolLayer style={{iconImage:'foo'}} />
   <Images>
     <Image key="foo">
       <View style={{backgroundColor:'red', width: 20, height: 20, borderRadius: 20}} />
     </Image>
   </Images>
   ```

#### Why?

As this is not supported iOS, also the alternative is more flexible.

<h3 id="source-layers-existing" style="{marginTop: 60px}">Implicit reference to existing source/layers:</h3>

#### Deprecated:
```jsx
<Mapbox.FillLayer id="building" style={{ visibility }} />
<Mapbox.LineLayer id="building-outline" style={{ visibility }} />

<Mapbox.VectorSource id="building" onPress={...} />
```

#### Replacement:
```jsx
<Mapbox.FillLayer id="building" existing style={{ visibility }} />
<Mapbox.LineLayer id="building-outline" existing style={{ visibility }} />

<Mapbox.VectorSource id="building" existing onPress={...} />
```

#### Why:
Vectors/Sources needs to wait for the id to appear if they marked as existing. Regular layers don't have to wait.

**Comment by KiwiKilian** [comment]
@mfazekas I guess you are missing the `existing` on the sources recommendation?

**Reply by mfazekas** [reply]
@KiwiKilian yep, thanks addressed
